### DISCUSSION TITLE
Static error codes

### DISCUSSION THREAD

**Discussion by alchar** [panva/jose#170] [Ideas]
It'd be great if `JOSEError`, and the errors that extend it, had static error codes. 

For example:
```
export class JOSEAlgNotAllowed extends JOSEError {
  static code = 'ERR_JOSE_ALG_NOT_ALLOWED'
  code = 'ERR_JOSE_ALG_NOT_ALLOWED'
}
```

This is just to make it possible to get the code, without having to create a new instance of the class. 

In our use case, we emit a metric with the error code on some failures. The unit tests look like:
```
assert(emittedMetric === new JWTExpired("test message").code);
```
which isn't nearly as clean as:
```
assert(emittedMetric === JWTExpired.code);
```

I imagine there are other cases where developers may find this a cleaner solution.

I'd be happy to submit this PR if it's something you'd like it. Either way, thanks for the great library!

**Comment by panva** [comment]
Sounds good. Would you give the PR a shot?

**Reply by alchar** [reply]
Will do!

**Reply by panva** [reply]
Altho the point of these error codes is that you just hard code their values.

You can just as well do instanceof checks otherwise.

**Reply by alchar** [reply]
We can definitely hardcode the values, I just found that to not be as clean as importing them directly from the library, especially if the string is used in multiple places.

`instanceof` checks work and are preferred when there's an object, but there isn't in this case.

If you're interested, I've opened the PR [here](https://github.com/panva/jose/pull/171).
